src/lib/fishingContest.ts

```diff
@@ -0,0 +1,331 @@
+		],
+		bait: getOSItem('Snowy knight')
+	},
+	{
+		id: 8,
+		name: 'East Piscarillius Ocean',
+		temperature: 'warm',
+		water: 'ocean',
+		specialtyVerb: 'firm',
+		allFishNames: [],
+		modifiers: [
+			{
+				name: 'Elite Kourend Diary',
+				has: async user => {
+					return (await userhasDiaryTier(user, KourendKebosDiary.elite))[0];
+				},
+				percentAddedToMin: 5
+			}
+		],
+		bait: getOSItem('Chinchompa')
+	},
+	// Lakes
+	{
+		id: 10,
+		name: 'Lake Molch',
+		temperature: 'cold',
+		water: 'lake',
+		specialtyVerb: 'winged',
+		allFishNames: [],
+		modifiers: [
+			{
+				name: 'Golden Tench',
+				has: async user => {
+					return user.cl().has('Golden tench');
+				},
+				percentAddedToMin: 5
+			}
+		],
+		bait: getOSItem('Orange feather')
+	},
+	{
+		id: 11,
+		name: 'Mort Myre Swamp Lake',
+		temperature: 'cold',
+		water: 'lake',
+		specialtyVerb: 'skeletal',
+		allFishNames: [],
+		modifiers: [
+			{
+				name: 'Elite Morytania Diary',
+				has: async user => {
+					return (await userhasDiaryTier(user, MorytaniaDiary.elite))[0];
+				},
+				percentAddedToMin: 5
+			}
+		],
+		bait: getOSItem('Mort myre fungus')
+	},
+	{
+		id: 12,
+		name: 'Wilderness Lava Lake',
+		temperature: 'warm',
+		water: 'lake',
+		specialtyVerb: 'magma',
+		allFishNames: [],
+		modifiers: [
+			{
+				name: 'Elite Wilderness Diary',
+				has: async user => {
+					return (await userhasDiaryTier(user, WildernessDiary.elite))[0];
+				},
+				percentAddedToMin: 5
+			}
+		],
+		bait: getOSItem('Raw lava eel')
+	}
+];
+
+for (const location of fishingLocations) {
+	for (const verb of [...verbMap[location.temperature], location.specialtyVerb]) {
+		for (const noun of waterMap[location.water]) {
+			location.allFishNames.push(`${toTitleCase(verb)} ${toTitleCase(noun)}`);
+		}
+	}
+}
+
+interface FishType {
+	temperature: Temperature;
+	water: Water;
+}
+
+// const allVerbs = [...fishingLocations.map(i => i.specialtyVerb), ...coldVerbs, ...warmVerbs];
+// const allNouns = [...lakeNouns, ...riverNouns, ...oceanNouns];
+
+export function getCurrentFishType(dateOverride?: Date): FishType {
+	const day = (dateOverride ?? new Date()).toLocaleDateString();
+
+	return {
+		temperature: murMurSort(['cold', 'warm'], day)[0],
+		water: murMurSort(['ocean', 'lake', 'river'], day)[0]
+	};
+}
+
+export function getValidLocationsForFishType(type: FishType) {
+	return fishingLocations.filter(loc => loc.temperature === type.temperature && loc.water === type.water);
+}
+
+export async function getUsersFishingContestDetails(user: KlasaUser) {
+	const catchesFromToday: FishingContestCatch[] = await prisma.$queryRawUnsafe(`SELECT *
+FROM fishing_contest_catch
+WHERE user_id = ${user.id}::bigint
+AND date::date = '${ISODateString()}'
+LIMIT 10;`);
+	const catchesAllTime = await prisma.fishingContestCatch.count({
+		where: {
+			user_id: BigInt(user.id)
+		}
+	});
+	const totalLength: [{ sum: number }] = await prisma.$queryRawUnsafe(`SELECT COALESCE(SUM(length_cm), 0) AS sum
```

Comment:
this should be `SUM(COALESCE(length_cm, 0)) as sum`

otherwise
10
null
11
will sum to null, and be coalesced to 0, instead of 21.